### PR TITLE
HTCONDOR-1835 branch

### PR DESCRIPTION
**Pull request by johnkn**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_schedd.V6/qmgmt_send_stubs.cpp [hunk 1]
```diff
@@ -125,14 +125,41 @@ NewCluster()
...
+
+			// Older versions of the SCHEDD do not return an error reply ad
+			// To handle that, look at what's on the wire, rather than what we should expect.
+			ClassAd reply;
+			bool gotClassad = false;
+			if ( ! qmgmt_sock->peek_end_of_message()) {
+				gotClassad = getClassAd(qmgmt_sock, reply);
+			}
+			if ( ! qmgmt_sock->end_of_message() && ! terrno) { // prefer existing errno
+				errno = ETIMEDOUT;
```
Comment (JaimeFrey) [inline src/condor_schedd.V6/qmgmt_send_stubs.cpp:138]: Setting of errno should happen right before we return. Intervening function calls may overwrite it.

**Review by JaimeFrey — CHANGES_REQUESTED** [review]
Looks good other than the errno handling.

File: src/condor_schedd.V6/qmgmt_send_stubs.cpp (hunk 1 above)
Comment (johnkn) [reply src/condor_schedd.V6/qmgmt_send_stubs.cpp]: fixed

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**